File: cardproject/Deck.java
package solitaire;
import java.util.Random;

public class Deck {
	private Card[] storage;
	private int top; 
	
	public Deck() {
		char [] suits = {'H','D','S','C'};
		char [] ranks = {'A','2','3','4','5','6','7','8','9','T','J','Q','K'};
		storage = new Card[52];
		top = 0;
		
		int count = 0;
		Card C1;
		for (int s = 0; s < suits.length; s++)
			for (int r = 0; r < ranks.length; r++) {
				C1 = new Card(ranks[r], suits[s]);
				storage[count++] = C1;
			}
		}
	
	public void display() {
		int x = 0;
		for (int i = 0; i < 4; i++) {
			for (int z = 0; z<13; z++) {
				storage[x++].display();
				System.out.print(" ");
			}
		System.out.println("");
		}
	}
	
	public void shuffle() { 
		Random rand = new Random();
		Card temp;
		
		int x, y;
		for (int i = 0; i<10000; i++) {
			x = rand.nextInt(52);
			y = rand.nextInt(52);
			temp = storage[x];
			storage[x] = storage[y];
			storage[y] = temp;
			
			
		}
	}
	
	public Card deal() {
		return storage[top++];
	}
	
	public int cardsLeft() { 
		return (52 - top);
	}

	}

File: lab_2/cards/Deck.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>
#include <string>
#include <time.h>

using namespace std;


void shuffleDeck(int deck[]) {

    //initialize random seed:
    srand(time(NULL));
    int original, swap, temp;

    for (int i = 0; i < 1000; i++) {
        original = rand() % 52;
        swap = rand() % 52;

        temp = deck[original];        
        deck[original] = deck[swap];
        deck[swap] = temp;       

    }
    
}

void displayCards(int deck[], string suits[], string ranks[]) {
    
    for (int i = 0; i < 4; i++) {
        int card = deck[i];
        int suitIndex = card / 13;
        int rankIndex = card % 13;

        cout << ranks[rankIndex] << " of " << suits[suitIndex] << endl;
    }
}

int main() {
    const int NUMBER_OF_CARDS = 52;
    int deck[NUMBER_OF_CARDS];
    string suits[] = {"Spades", "Hearts", "Diamonds", "Clubs"};
    string ranks[] = {"Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King"};

    //initialize cards
    for (int i = 0; i < NUMBER_OF_CARDS; i++)
        deck[i] = i;

    //Shuffle the cards
    for (int i = 0; i < NUMBER_OF_CARDS; i++)
        cout << deck[i] << " ";

    cout << endl;
    cout << endl;
    shuffleDeck(deck);

    for (int i = 0; i < NUMBER_OF_CARDS; i++)
        cout << deck[i] << " ";

    cout << endl;




    //Generate random index




    //Display first four cards
    displayCards(deck, suits, ranks);


    return 0;
}
